lms/security.py

```diff
@@ -48,9 +48,51 @@ def get_policy(request):
         return LMSGoogleSecurityPolicy()
 
     if path in {"/lti_launches", "/content_item_selection", "/api/gateway/h/lti"}:
+        # Actual LTI backed authentication
         return LTIUserSecurityPolicy(get_lti_user_from_launch_params)
 
-    return LTIUserSecurityPolicy(get_lti_user)
+    if path.startswith("/api") and path.endswith("callback"):
+        # LTIUser serialized in the state param for the oauth flow
+        return LTIUserSecurityPolicy(get_lti_user_from_oauth_callback)
+
+    if path.startswith("/api") and path.endswith("authorize"):
+        # LTUser serialized as query param for authorization failures
+        return LTIUserSecurityPolicy(
+            partial(get_lti_user_from_bearer_token, location="querystring")
+        )
+
+    if path.startswith("/api"):
+        # LTUser serialized in the headers for API calls from the frontend
+        return LTIUserSecurityPolicy(
+            partial(get_lti_user_from_bearer_token, location="headers")
+        )
+
+    if path in {"/assignment"}:
+        # LTUser serialized in a from for non deep-linked assignment configuration
+        return LTIUserSecurityPolicy(
```

Comment:
We add that to the form we send to ourselves (via frontend) on configuration here:

https://github.com/hypothesis/lms/blob/main/lms/views/lti/basic_launch.py#L103